lib/syscall_shim/include/uk/syscall.h

```diff
@@ -44,14 +44,48 @@
 #include <uk/errptr.h>
 #include <errno.h>
 #include <stdarg.h>
+#include <stddef.h>
 #include <uk/print.h>
 #include <uk/legacy_syscall.h>
 #include <uk/bits/syscall_linuxabi.h>
+#include <uk/syscall_exittab.h>
+#include <uk/syscall_entertab.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+#if CONFIG_LIBSYSCALL_SHIM
+static inline void uk_syscall_enter_handler(struct uk_syscall_enter_ctx *sctx)
+{
+	/* Mark system call entry */
+	uk_syscall_nested_depth++;
+
+	/* Only run this if the first system call in the callchain */
+	if (uk_syscall_nested_depth != 1)
```

Comment:
I'm thinking whether it's more flexible to pass the nested depth as a parameter to the handlers and let them decide whether they should be running on nested syscalls or not.